internal/host/validator.go

```diff
@@ -417,271 +389,211 @@ func (v *validator) areTangServersReachable(c *validationContext) ValidationStat
-	return boolValue(len(disks) > 0)
-}
-
-func (v *validator) printHasMinValidDisks(c *validationContext, status ValidationStatus) string {
-	switch status {
-	case ValidationSuccess:
-		return "Sufficient disk capacity"
-	case ValidationFailure:
-		return "No eligible disks were found, please check specific disks to see why they are not eligible"
-	case ValidationPending:
-		return "Missing inventory"
-	default:
-		return fmt.Sprintf("Unexpected status %s", status)
+	if boolValue(len(disks) > 0) == ValidationSuccess {
+		return ValidationSuccess, "Sufficient disk capacity"
 	}
+	return ValidationFailure, "No eligible disks were found, please check specific disks to see why they are not eligible"
 }
 
-func (v *validator) isMachineCidrDefined(c *validationContext) ValidationStatus {
+func (v *validator) isMachineCidrDefined(c *validationContext) (ValidationStatus, string) {
+	status := ValidationSuccessSuppressOutput
 	if c.infraEnv != nil {
-		return ValidationSuccessSuppressOutput
+		return status, fmt.Sprintf("Unexpected status %s", status)
 	}
-	return boolValue(swag.BoolValue(c.cluster.UserManagedNetworking) || swag.StringValue(c.cluster.Kind) == models.ClusterKindAddHostsCluster || network.IsMachineCidrAvailable(c.cluster))
-}
-
-func (v *validator) printIsMachineCidrDefined(context *validationContext, status ValidationStatus) string {
-	switch status {
-	case ValidationSuccess:
-		if swag.BoolValue(context.cluster.UserManagedNetworking) {
-			return "No Machine Network CIDR needed: User Managed Networking"
-		}
-		if swag.StringValue(context.cluster.Kind) == models.ClusterKindAddHostsCluster {
-			return "No Machine Network CIDR needed: Day2 cluster"
+	status = boolValue(swag.BoolValue(c.cluster.UserManagedNetworking) || swag.StringValue(c.cluster.Kind) == models.ClusterKindAddHostsCluster || network.IsMachineCidrAvailable(c.cluster))
+	if status == ValidationSuccess {
+		if swag.BoolValue(c.cluster.UserManagedNetworking) {
+			return ValidationSuccess, "No Machine Network CIDR needed: User Managed Networking"
 		}
-		return "Machine Network CIDR is defined"
-	case ValidationFailure:
-		if swag.BoolValue(context.cluster.VipDhcpAllocation) {
-			return "Machine Network CIDR is undefined"
-		} else {
-			return "Machine Network CIDR is undefined; the Machine Network CIDR can be defined by setting either the API or Ingress virtual IPs"
+		if swag.StringValue(c.cluster.Kind) == models.ClusterKindAddHostsCluster {
+			return ValidationSuccess, "No Machine Network CIDR needed: Day2 cluster"
 		}
-	default:
-		return fmt.Sprintf("Unexpected status %s", status)
+		return ValidationSuccess, "Machine Network CIDR is defined"
 	}
+	if swag.BoolValue(c.cluster.VipDhcpAllocation) {
+		return ValidationFailure, "Machine Network CIDR is undefined"
+	}
+	return ValidationFailure, "Machine Network CIDR is undefined; the Machine Network CIDR can be defined by setting either the API or Ingress virtual IPs"
```

Comment:
This whole middle bit of the function could be better I think. A lot of the conditions are duplicated. I'm pretty sure this is identical.

```go
	if swag.BoolValue(c.cluster.UserManagedNetworking) {
		return ValidationSuccess, "No Machine Network CIDR needed: User Managed Networking"
	}
	if swag.StringValue(c.cluster.Kind) == models.ClusterKindAddHostsCluster {
		return ValidationSuccess, "No Machine Network CIDR needed: Day2 cluster"
	}
	if network.IsMachineCidrAvailable(c.cluster) {
		return ValidationSuccess, "Machine Network CIDR is defined"
	}
	if swag.BoolValue(c.cluster.VipDhcpAllocation) {
		return ValidationFailure, "Machine Network CIDR is undefined"
	}
	return ValidationFailure, "Machine Network CIDR is undefined; the Machine Network CIDR can be defined by setting either the API or Ingress virtual IPs"
```